test/gradcheck.jl

```diff
@@ -174,6 +174,11 @@ end
 @test gradient(det, 2.0)[1] == 1
 @test gradient(logdet, 2.0)[1] == 0.5
 
+
+@test gradient(x -> sum(x .== 0.5), rand(3, 3))[1] === nothing
+x1 = rand(3, 3)
```

Comment:
I'm not sure if this is a good practise in Julia. Or should I create a `testset` put this under that?

Reply:
It's a bit of a free-for-all. Maybe there's an existing broadcasting testset these could join? 

It might be worth adding a comment linking to the issue, as some of the others have.